src/global_properties.rs

```diff
@@ -2,11 +2,53 @@ use crate::context::Context;
 use crate::error::IxaError;
 use serde::de::DeserializeOwned;
 use std::any::{Any, TypeId};
+use std::cell::RefCell;
 use std::collections::HashMap;
 use std::fmt::Debug;
 use std::fs;
 use std::io::BufReader;
 use std::path::Path;
+use std::sync::Arc;
+use std::sync::LazyLock;
+use std::sync::Mutex;
+
+type PropertySetterFn =
+    dyn Fn(&mut Context, &str, serde_json::Value) -> Result<(), IxaError> + Send + Sync;
+
+#[allow(clippy::type_complexity)]
+pub static GLOBAL_PROPERTIES: LazyLock<Mutex<RefCell<HashMap<String, Arc<PropertySetterFn>>>>> =
```

Comment:
This could probably use some explanation in a comment

Reply:
CoPilot says :"The GLOBAL_PROPERTIES variable is a static global variable that holds a thread-safe, lazily-initialized, and mutable reference to a HashMap. The HashMap maps string keys to Arc-wrapped function pointers (PropertySetterFn). This variable is defined using the LazyLock type to ensure it is only initialized once and is safe to access from multiple threads. The Mutex ensures mutual exclusion for modifying the HashMap, while the RefCell allows for interior mutability, enabling modifications to the HashMap even when it is accessed through a shared reference."

Reply:
I'll obviously clean that up.